core/domain/exp_services.py

```diff
@@ -1021,11 +1021,8 @@ def _save_exploration(
             stored exploration model do not match.
     """
     exploration_rights = rights_manager.get_exploration_rights(exploration.id)
-    exploration.validate(
-        strict=(
-            exploration_rights.status != rights_domain.ACTIVITY_STATUS_PRIVATE
-        )
-    )
+    status = exploration_rights.status != rights_domain.ACTIVITY_STATUS_PRIVATE
```

Comment:
It is misleading to call this variable "status". It's a boolean, whereas a status is an enum.

Do something like exploration_is_public or exploration_is_private, perhaps. Please try to name your variables so that they clearly describe the meaning of what is being stored (otherwise, there will be more back-and-forth in code review).

Reply:
changed to exploration_is_public